ecs-agent/tmds/handlers/v2/response.go

```diff
@@ -0,0 +1,95 @@
+// permissions and limitations under the License.
+package v2
+
+import (
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/tmds/handlers/response"
+)
+
+// TaskResponse defines the schema for the task response JSON object
+type TaskResponse struct {
+	Cluster               string              `json:"Cluster"`
+	TaskARN               string              `json:"TaskARN"`
+	Family                string              `json:"Family"`
+	Revision              string              `json:"Revision"`
+	DesiredStatus         string              `json:"DesiredStatus,omitempty"`
+	KnownStatus           string              `json:"KnownStatus"`
+	Containers            []ContainerResponse `json:"Containers,omitempty"`
+	Limits                *LimitsResponse     `json:"Limits,omitempty"`
+	PullStartedAt         *time.Time          `json:"PullStartedAt,omitempty"`
+	PullStoppedAt         *time.Time          `json:"PullStoppedAt,omitempty"`
+	ExecutionStoppedAt    *time.Time          `json:"ExecutionStoppedAt,omitempty"`
+	AvailabilityZone      string              `json:"AvailabilityZone,omitempty"`
+	TaskTags              map[string]string   `json:"TaskTags,omitempty"`
+	ContainerInstanceTags map[string]string   `json:"ContainerInstanceTags,omitempty"`
+	LaunchType            string              `json:"LaunchType,omitempty"`
+	Errors                []ErrorResponse     `json:"Errors,omitempty"`
+}
+
+// ContainerResponse defines the schema for the container response
+// JSON object
+type ContainerResponse struct {
+	ID            string                    `json:"DockerId"`
+	Name          string                    `json:"Name"`
+	DockerName    string                    `json:"DockerName"`
+	Image         string                    `json:"Image"`
+	ImageID       string                    `json:"ImageID"`
+	Ports         []response.PortResponse   `json:"Ports,omitempty"`
+	Labels        map[string]string         `json:"Labels,omitempty"`
+	DesiredStatus string                    `json:"DesiredStatus"`
+	KnownStatus   string                    `json:"KnownStatus"`
+	ExitCode      *int                      `json:"ExitCode,omitempty"`
+	Limits        LimitsResponse            `json:"Limits"`
+	CreatedAt     *time.Time                `json:"CreatedAt,omitempty"`
+	StartedAt     *time.Time                `json:"StartedAt,omitempty"`
+	FinishedAt    *time.Time                `json:"FinishedAt,omitempty"`
+	Type          string                    `json:"Type"`
+	Networks      []response.Network        `json:"Networks,omitempty"`
+	Health        *HealthStatus             `json:"Health,omitempty"`
+	Volumes       []response.VolumeResponse `json:"Volumes,omitempty"`
+	LogDriver     string                    `json:"LogDriver,omitempty"`
+	LogOptions    map[string]string         `json:"LogOptions,omitempty"`
+	ContainerARN  string                    `json:"ContainerARN,omitempty"`
+}
+
+// Container health status
+type HealthStatus struct {
```

Comment:
This is a new type which matches the `apicontainer.HealthStatus` type that is being used in v2 metadata response currently. 

 https://github.com/aws/amazon-ecs-agent/blob/883211e24d9255023a68f6351001c291af8a699b/agent/api/container/container.go#L111-L120

This is being done to decouple metadata response models from API models.